solace-java-spring-boot-autoconfigure/src/main/java/com/solace/labs/spring/boot/autoconfigure/SolaceJavaProperties.java

```diff
@@ -86,6 +90,35 @@
      * @see <a href="http://docs.solace.com/Solace-Messaging-APIs/Configuring-Connection-T.htm">http://docs.solace.com/Solace-Messaging-APIs/Configuring-Connection-T.htm</a>
      */
     private int reconnectRetryWaitInMillis = 3000;
+
+    /**
+     * Indicates whether the acknowledgement of a received guaranteed message is done by the
+     * the API or by application explicitly calling ackMessage() on the received message
+     * <p>
+     * Possible values are: <p>
+     * <ul>
+     * <li>auto_ack</li>
+     * <li>client_ack</li>
+     * </ul>
+     * Default value: auto_ack
+     */
+    private String messageAckMode = "auto_ack";
+
+    /**
+     * If enabled, the API maintains a local cache of subscriptions and reapplies
+     * them when the subscriber connection is reestablished.
+     * <p>
+     * Reapply subscriptions will only apply direct topic subscriptions unpon a Session reconnect.
+     * It will not reapply topic subscriptions on durable and non-durable endpoints
+     * <p>
+     * Default: false
+     */
+    private boolean reapplySubscriptions = false;
+
+    /**
+     * Reserved attributes used by internal clients of the messaging API
+     */
+    private Properties reserved = new Properties();
```

Comment:
Agreed, also added references to fromProperties and toProperties in JCSMPProperties